assets/base.css

```diff
@@ -1216,86 +1216,111 @@ deferred-media {
+  --shadow-blur-radius: var(--buttons-shadow-blur-radius);
+  --shadow-opacity: var(--buttons-shadow-opacity);
+  --border-offset: var(--buttons-border-offset); /* reduce radius edge artifacts */
+  --border-opacity: calc(1 - var(--buttons-border-opacity));
+  border-radius: var(--buttons-radius);
+  position: relative;
+  min-width: calc(12rem + var(--buttons-border-width) * 2);
+  min-height: calc(4.5rem + var(--buttons-border-width) * 2);
+}
+
+.shopify-payment-button__button--branded {
+  z-index: auto;
+}
+
+.cart__dynamic-checkout-buttons iframe {
+  box-shadow: var(--shadow-horizontal-offset) var(--shadow-vertical-offset) var(--shadow-blur-radius) rgba(var(--color-base-text), var(--shadow-opacity));
+}
+
 .button,
 .shopify-challenge__button,
 .customer button {
-  cursor: pointer;
   display: inline-flex;
   justify-content: center;
   align-items: center;
-  box-sizing: border-box;
+  border: 0;
+  padding: 0 3rem;
+  border-radius: var(--buttons-radius);
+  cursor: pointer;
   font: inherit;
   font-size: 1.5rem;
-  padding: 0.9rem 3rem 1.1rem;
   text-decoration: none;
-  border: 1px solid transparent;
-  background-color: rgba(var(--color-button), var(--alpha-button-background));
   color: rgb(var(--color-button-text));
-  min-width: 12rem;
-  min-height: 4.5rem;
+  transition: box-shadow var(--duration-short) ease;
   -webkit-appearance: none;
   appearance: none;
+  background-color: rgba(var(--color-button), var(--alpha-button-background));
 }
 
-.button,
-.shopify-challenge__button,
-.customer button,
-.shopify-payment-button__button--unbranded,
-.shopify-payment-button [role="button"] {
-  box-shadow: 0 0 0 var(--buttons-border-width) rgba(var(--color-button), var(--buttons-border-opacity));
+.button:before,
+.shopify-challenge__button:before,
+.customer button:before,
+.shopify-payment-button__button--unbranded:before,
+.shopify-payment-button [role="button"]:before,
+.cart__dynamic-checkout-buttons [role='button']:before {
+  content: '';
+  position: absolute;
+  inset: 0;
+  z-index: -1;
+  border-radius: var(--buttons-radius);
+  box-shadow: var(--shadow-horizontal-offset) var(--shadow-vertical-offset) var(--shadow-blur-radius) rgba(var(--color-base-text), var(--shadow-opacity));
 }
 
-.button,
-.shopify-challenge__button,
-.customer button,
-.shopify-payment-button__button--unbranded,
-.shopify-payment-button [role="button"],
-.cart__dynamic-checkout-buttons [role='button'],
-.cart__dynamic-checkout-buttons iframe {
-  --shadow-horizontal-offset: var(--buttons-shadow-horizontal-offset);
-  --shadow-vertical-offset: var(--buttons-shadow-vertical-offset);
-  --shadow-blur-radius: var(--buttons-shadow-blur-radius);
-  --shadow-opacity: var(--buttons-shadow-opacity);
+.button:after,
+.shopify-challenge__button:after,
+.customer button:after,
+.shopify-payment-button__button--unbranded:after {
+  content: '';
+  position: absolute;
+  inset: var(--buttons-border-width);
```

Comment:
This seem to mess with higher value. Because we're applying a radius on something that isn't the same size as the actual element, it ends up having a different "curve" to it. So when you go on higher values like `24px` then you notice it even more: 
[not super high value](https://screenshot.click/21-12-eydor-c5dg7.png)
[higher value with not so high border-radius](https://screenshot.click/21-12-ppyh6-8ewox.png)

Reply:
Good find. This was happening at ranges I wasn't necessarily testing with much. It's a case of needing to add the radius and the border width. Should be fixed but let me know if you can still break it.